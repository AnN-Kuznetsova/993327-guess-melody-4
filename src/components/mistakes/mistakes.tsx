import * as React from "react";


interface Props {
  count: number;
};


export const Mistakes: React.FunctionComponent<Props> = (props: Props) => {
  const {count} = props;

  const mistakes = new Array(count).fill(``);

  return (
    <div className="game__mistakes">
      {mistakes.map((mistake, index) => <div key={`mistake-${index}`} className="wrong" />)}
    </div>
  );
};
