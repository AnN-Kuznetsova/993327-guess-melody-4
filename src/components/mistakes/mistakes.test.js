import React from "react";
import renderer from "react-test-renderer";
import {Mistakes} from "./mistakes";


describe(`Render Mistakes`, () => {
  it(`Render Mistakes should match with snapshot with zero count`, () => {
    const mistakesSnapshot = renderer.create(
        <Mistakes count={0} />
    ).toJSON();

    expect(mistakesSnapshot).toMatchSnapshot();
  });


  it(`Render Mistakes should match with snapshot with one count`, () => {
    const mistakesSnapshot = renderer.create(
        <Mistakes count={1} />
    ).toJSON();

    expect(mistakesSnapshot).toMatchSnapshot();
  });
});
