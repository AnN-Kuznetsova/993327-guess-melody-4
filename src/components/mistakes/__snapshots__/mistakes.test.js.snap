// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Mistakes Render Mistakes should match with snapshot with one count 1`] = `
<div
  className="game__mistakes"
>
  <div
    className="wrong"
  />
</div>
`;

exports[`Render Mistakes Render Mistakes should match with snapshot with zero count 1`] = `
<div
  className="game__mistakes"
/>
`;
